overlord/devicestate/devicemgr.go

```diff
@@ -309,16 +309,22 @@ func (m *DeviceManager) StartUp() error {
 	return EarlyConfig(m.state, m.preloadGadget)
 }
 
-func (m *DeviceManager) maybeSetupUbuntuSave() error {
-	// only called for UC20
+func (m *DeviceManager) shouldMountUbuntuSave() bool {
+	if release.OnClassic {
+		return false
+	}
+	// TODO:UC20+: ubuntu-save needs to be mounted for recover too
+	return m.SystemMode(SysHasModeenv) == "install" ||
```

Comment:
I think adding install here adds extra work that will end up doing nothing, as this is used from StartUp which is too early in install anyway to have an available save

Reply:
Ah, right, thats a good point